test/e2e/farm_build_test.go

```diff
@@ -0,0 +1,1379 @@
+			tag    string
+		}
+
+		type build struct {
+			arch                string
+			expectedTobeBuiltOn string
+			usingEmulation      bool
+			withCleanup         bool
+		}
+
+		type expectBuildsOf []build
+
+		type expectFailureWith struct {
+			message string
+		}
+
+		/*##########################################################################################################*/
+
+		var runRegistry = func() string {
+			// Set up a registry server running locally in a container. This has to be accessible both
+			// locally and from our secondary podman servers.
+
+			// NB Very occasionally, this port seems to get locked up and is not released when the test completes, meaning
+			// all subsequent tests start failing. I'm not sure why, but a reboot of the VM will sort this out.
+			lock := GetPortLock("5002")
+			defer lock.Unlock()
+
+			regName := "someRandomReg"
+
+			cmd := exec.Command(podmanStaticTest.PodmanBinary, "run", "-d", "--replace", "--rm", "--name", regName, "-p", "5002:5000", REGISTRY_IMAGE, "/entrypoint.sh", "/etc/docker/registry/config.yml")
+
+			session, err := Start(cmd, GinkgoWriter, GinkgoWriter)
+
+			// I think this waiting 1 sec for listening on to appear in the log, and then repeating 20x
+			// if !WaitContainerReady(podmanStaticTest, "registry", "listening on", 20, 1) {
+			//    Skip("Cannot start docker registry.")
+			// }
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("%q failed to execute", podmanStaticTest.PodmanBinary))
+			Eventually(session, DefaultWaitTimeout).Should(Exit(0))
+
+			return regName
+		}
+		/*##########################################################################################################*/
+
+		var stopRegistry = func(regName string) {
+			cmd := exec.Command(podmanStaticTest.PodmanBinary, "stop", regName)
+
+			session, err := Start(cmd, GinkgoWriter, GinkgoWriter)
+
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("%q failed to execute", podmanStaticTest.PodmanBinary))
+			Eventually(session, DefaultWaitTimeout).Should(Exit(0))
+		}
+		/*##########################################################################################################*/
+
+		var skopeo = func(image string) *PodmanSessionIntegration {
+
+			// skopeo inspect --tls-verify=false  --raw docker://img.pegortech.co.uk/fedora would typically produce output:
+			//
+			// {"schemaVersion":2,"mediaType":"application/vnd.oci.image.index.v1+json","manifests":[{"mediaType":"appli\
+			// cation/vnd.oci.image.manifest.v1+json","digest":"sha256:6dbd1777b7dfa0d6b595d3220cc05923eb9833d6e49263b65\
+			// 32c0ea182765017","size":612,"platform":{"architecture":"arm64","os":"linux","variant":"v8"}},{"mediaType"\
+			// :"application/vnd.oci.image.manifest.v1+json","digest":"sha256:6afa31388822c02a0452e190617affa6257d3c28ac5\
+			// 4681ab0ab35193ab86df3","size":680,"platform":{"architecture":"amd64","os":"linux"}}]}
+
+			var SkopeoBinary = "skopeo"
+
+			var dockerImage = fmt.Sprintf("docker:%s%s", `//`, image)
```

Comment:
why the double %s%s? just write it as
`fmt.Sprintf("docker://%s", image)`

and also the var name = value syntax is generally not very go style. please use
`name := value` if you assign a value right away